### PR TITLE
PAINT-93: Gradle Code Coverage Plugin

### PR DESCRIPTION
**Pull request by tanvidadu**

### CODE REVIEW

**Comment by ThomasSchwengler** [comment]
test this please

File: Paintroid/build.gradle [hunk 1]
```diff
@@ -16,6 +16,8 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+apply plugin: 'com.android.application'
```
Comment (ThomasSchwengler) [inline Paintroid/build.gradle]: Please remove duplicated plugin entry

File: build.gradle [hunk 2]
```diff
@@ -23,7 +23,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.3'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (ThomasSchwengler) [inline build.gradle]: Is it necessary to upgrade the android plugin to 3.0.1 for this plugin?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,24 +1,6 @@
...
-#  This program is distributed in the hope that it will be useful,
-#  but WITHOUT ANY WARRANTY; without even the implied warranty of
-#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-#  GNU Affero General Public License for more details.
-#
-#  You should have received a copy of the GNU Affero General Public License
-#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-
+#Sun Dec 31 07:52:13 IST 2017
```
Comment (ThomasSchwengler) [inline gradle/wrapper/gradle-wrapper.properties]: Please don't remove the copyright header

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,24 +1,6 @@
...
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (ThomasSchwengler) [inline gradle/wrapper/gradle-wrapper.properties]: Is it necessary to upgrade gradle for this plugin to work?

**Comment by ThomasSchwengler** [comment]
Hello!
Can you please rename the commit to "PAINT-93: Gradle Code Coverage Plugin"?

File: Paintroid/build.gradle (hunk 1 above)
Comment (tanvidadu) [reply Paintroid/build.gradle]: will do

File: build.gradle (hunk 2 above)
Comment (tanvidadu) [reply build.gradle]: Not necessary will revert back to android plugin to 2.3.3

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (tanvidadu) [reply gradle/wrapper/gradle-wrapper.properties]: not necessary will revert back to original version .

**Comment by tanvidadu** [comment]
I have taken care of all the mentioned concerns above . 
plz review the changes @ThomasSchwengler 
.

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -16,7 +16,7 @@
 #  You should have received a copy of the GNU Affero General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-
+#Sun Dec 31 07:52:13 IST 2017
```
Comment (ThomasSchwengler) [inline gradle/wrapper/gradle-wrapper.properties]: remove this line please

File: Paintroid/build.gradle [hunk 6]
```diff
@@ -17,7 +17,9 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+
```
Comment (ThomasSchwengler) [inline Paintroid/build.gradle]: Remove this line please

**Comment by ThomasSchwengler** [comment]
Can you squash your commits into one?
Thank you

**Comment by tanvidadu** [comment]
conflicts are occurring while trying to squash it into one commit , with the  Gradle Code Coverage Plugin fixed Jacoco added commit. Plz advise on how to do it.

**Comment by ThomasSchwengler** [comment]
If you set https://github.com/Catrobat/Paintroid as `upstream` this should look somewhat like this:
`git remote update` - This updates your remote repositories
`git reset upstream/develop` - Keeps your file changes but will remove any commits
`git commit -a -m "PAINT-93: Gradle Code Coverage Plugin"` - Creates a new (single) commit

Have a look at [this link](https://makandracards.com/makandra/527-squash-several-git-commits-into-a-single-commit) if you want to read more about this.

**Comment by ThomasSchwengler** [comment]
Hi again, you have been added on jira, please drag-and-drop your ticket to "In development" (and to "ready for code review" once you have rebased)
Also can you please rebase your commit?
Thanks :)

**Comment by tanvidadu** [comment]
Hi @ThomasSchwengler 
While rebasing the commit I am facing an merge conflict with Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/BaseToolWithRectangleShape.java
file. Plz advice me on how to solve this conflict.

**Comment by ThomasSchwengler** [comment]
Hi, since it's only 2 lines you are actually editing, you could just start over (It's of course possible to get this merge conflict resolved, but I guess it's easier to just start from scratch).
Just be sure to start from an up-to-date develop.

If you still want to rebase though, you could read yourself into https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase on how to rebase a commit.

**Comment by ThomasSchwengler** [comment]
Here is how you could do it:
https://streamable.com/717a7

In this video am discarding changes made to any files but `build.gradle` and `Paintroid/build.gradle` and create a new clean commit, which I then rebase on the develop.
I hope you get the idea, you can read more about git e.g. here https://git-scm.com/book/en/v2

**Comment by tanvidadu** [comment]
Thanks a ton @ThomasSchwengler !!
I have a made a noob error . I will immediately send a new PR.

**Pull request closed without merging**